src/brain/projectsHandler/project.ts

```diff
@@ -0,0 +1,67 @@
+import { EmitterWebhookEvent } from '@octokit/webhooks';
+import * as Sentry from '@sentry/node';
+
+import { ClientType } from '@/api/github/clientType';
+import { ISSUES_PROJECT_NODE_ID, PRODUCT_AREA_FIELD_ID } from '@/config';
+import { PRODUCT_AREA_LABEL_PREFIX } from '@/config';
+import { shouldSkip } from '@/utils/githubEventHelpers';
+import { getClient } from '@api/github/getClient';
+import {
+  getIssueDetailsFromNodeId,
+  getProductAreaFromProjectField,
+} from '@utils/githubEventHelpers';
+
+function isNotInAProjectWeCareAbout(payload) {
+  return payload?.projects_v2_item?.project_node_id !== ISSUES_PROJECT_NODE_ID;
+}
+
+function isNotAProjectFieldWeCareAbout(payload) {
+  return payload?.changes?.field_value?.field_node_id !== PRODUCT_AREA_FIELD_ID;
+}
+
+function isMissingNodeId(payload) {
+  return (
+    payload?.projects_v2_item?.node_id == null ||
+    payload?.projects_v2_item?.content_node_id == null
+  );
+}
+
+export async function syncLabelsWithProjectField({
+  id,
+  payload,
+  ...rest
+}: EmitterWebhookEvent<'projects_v2_item.edited'>) {
+  const tx = Sentry.startTransaction({
+    op: 'brain',
+    name: 'issueLabelHandler.syncLabelsWithProjectField',
+  });
+
+  const reasonsToDoNothing = [
+    isNotInAProjectWeCareAbout,
+    isNotAProjectFieldWeCareAbout,
+    isMissingNodeId,
+  ];
+  if (await shouldSkip(payload, reasonsToDoNothing)) {
+    return;
+  }
+
+  const owner = payload?.organization?.login || '';
```

Comment:
Coming here from #524 ... I'm wondering why we default to the empty string here. I doubt an `App` octokit loaded with an empty org would work. I think we should let an exception bubble up if `payload.organization.login` does not exist.